File: projeto_ed_para_praticar/repository/pilha/Empilhavel.java
package repository.pilha;

//Stackable
public interface Empilhavel {
	//métodos principais
	void empilhar(Object dado);	//C - push
	Object espiar();			//R - peek
	void atualizar(Object dado);//U - update
	Object desempilhar();		//D - pop
	
	//métodos auxiliares
	boolean estaCheia();	//isFull
	boolean estaVazia();	//isEmpty
	String imprimir();	    //print
}

File: pilha/dinamica/Kotlin/Pilha Genérica/Empilhavel.kt
interface Empilhavel<T> {
    // Metodos principais
    fun empilhar(dado: T?)
    fun desempilhar(): T?
    fun espiar(): T?
    fun atualizar(dado: T?)     
    // Metodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}